### PR TITLE
paging via continuation token

### PR DESCRIPTION
**Pull request by ArielElp**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44)

### CODE REVIEW

**Review by lior-stark — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @anatgstarkware and @ArielElp)

___
*[`api/starknet_api_openrpc.json` line 667 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/44#-N9RL8lk4Z3tItIVeGI2:-N9RL8lk4Z3tItIVeGI3:b-8rqeiu) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/cbc6c341ee987cda731330f688c9f0e81e5e26ae/api/starknet_api_openrpc.json#L667)):*
> ```JSON
>                     },
>                     "chunk_size": {
>                         "type": "integer"
> ```

shouldn't there be a minimum here?

**Review by anatgstarkware — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44)* status: all files reviewed (commit messages unreviewed), 2 unresolved discussions (waiting on @ArielElp)

___
*[`api/starknet_api_openrpc.json` line 612 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/44#-N9RUf2MEOoUoQhm-goQ:-N9RUf2MEOoUoQhm-goR:b-79fiyx) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/cbc6c341ee987cda731330f688c9f0e81e5e26ae/api/starknet_api_openrpc.json#L612)):*
> ```JSON
>                 }
>             },
>             "errors": [
> ```

I think we need INVALID_CONTINUATION_TOKEN as well

File: api/starknet_api_openrpc.json [hunk 1]
```diff
@@ -599,21 +599,13 @@
...
-                        "page_number": {
-                            "title": "Page Number",
-                            "description": "The returned page number.",
-                            "type": "integer"
-                        },
-                        "is_last_page": {
-                            "title": "Is Last Page",
-                            "description": "A flag indicating whether this is the end of the stream of events",
-                            "type": "boolean"
+                        "continuation_token": {
```
Comment (Mirko-von-Leipzig) [inline api/starknet_api_openrpc.json:602]: I think this might commonly be known as a `cursor`.

**Review by ArielElp — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44)* status: all files reviewed (commit messages unreviewed), 2 unresolved discussions (waiting on @lior-stark and @Mirko-von-Leipzig)

___
*[`api/starknet_api_openrpc.json` line 602 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/44#-N9RBYMVvSMrSwMj9cww-r1-602:-N9S1CwZ9HUE1ca687B_:b-eseif1) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/cbc6c341ee987cda731330f688c9f0e81e5e26ae/api/starknet_api_openrpc.json#L602)):*
<details><summary><i>Previously, Mirko-von-Leipzig wrote…</i></summary><blockquote>

I think this might commonly be known as a `cursor`.
</blockquote></details>

If this is the accepted terminology here I'll update it, but a quick search led to this argument which makes sense IMO:

 "Continuation Token is a better term than cursor at it avoids confusion with DB cursors"
___
*[`api/starknet_api_openrpc.json` line 612 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/44#-N9RUf2MEOoUoQhm-goQ:-N9S2YDq9zDngEmSFuvj:b-eriejq) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/cbc6c341ee987cda731330f688c9f0e81e5e26ae/api/starknet_api_openrpc.json#L612)):*
<details><summary><i>Previously, anatgstarkware (anatg) wrote…</i></summary><blockquote>

I think we need INVALID_CONTINUATION_TOKEN as well
</blockquote></details>

Done

Probably a cost of abstracting this away but it's a bit weird to see an error in the spec with no uniform semantics.
___
*[`api/starknet_api_openrpc.json` line 667 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/44#-N9RL8lk4Z3tItIVeGI2:-N9S2lG53LnedIDxGtUw:b-69uhs2) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/cbc6c341ee987cda731330f688c9f0e81e5e26ae/api/starknet_api_openrpc.json#L667)):*
<details><summary><i>Previously, lior-stark wrote…</i></summary><blockquote>

shouldn't there be a minimum here?
</blockquote></details>

hhmmm probably, not sure what's a reasonable magic number here but left it at 1 as was the restriction for page size previously

**Review by lior-stark — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44)* status: all files reviewed, 1 unresolved discussion (waiting on @Mirko-von-Leipzig)

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Review by anatgstarkware — APPROVED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44#-:-N9ap3Y-8R0FlvSGpaZY:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @Mirko-von-Leipzig)

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Review by ArielElp — COMMENTED** [review]
> Dismissed @Mirko-von-Leipzig from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44)* status: all files reviewed, 1 unresolved discussion (waiting on @anatgstarkware, @lior-stark, and @Mirko-von-Leipzig)

**Review by anatgstarkware — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/44)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ArielElp)

**Pull request merged**